File: src/main/java/pro/sky/java/weatherapp/handler/WeatherHandler.java
package pro.sky.java.weatherapp.handler;

import lombok.RequiredArgsConstructor;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import pro.sky.java.weatherapp.domain.Weather;
import pro.sky.java.weatherapp.service.WeatherService;
import reactor.core.publisher.Mono;

import static org.springframework.web.reactive.function.server.ServerResponse.ok;

@Component
@RequiredArgsConstructor
public class WeatherHandler {

    private final WeatherService service;

    @NonNull
    public Mono<ServerResponse> get(ServerRequest serverRequest) {
        String cityName = serverRequest.pathVariable("cityName");
        return ok().body(
                service.getWeatherByCity(cityName),
                Weather.class
        );
    }
}

File: src/main/kotlin/pro/sky/java/weatherapp/handler/WeatherHandler.kt
package pro.sky.java.weatherapp.handler

import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.server.ServerRequest
import org.springframework.web.reactive.function.server.ServerResponse
import org.springframework.web.reactive.function.server.ServerResponse.ok
import org.springframework.web.reactive.function.server.bodyValueAndAwait
import pro.sky.java.weatherapp.service.WeatherService

@Component
class WeatherHandler(
    private val service: WeatherService
) {
    suspend fun get(serverRequest: ServerRequest): ServerResponse {
        val cityName: String = serverRequest.pathVariable("cityName")
        return ok().bodyValueAndAwait(service.getWeatherByCity(cityName))
    }
}